src/BaGet.Decompiler.Nupkg/NugetDecompilerService.cs

```diff
@@ -0,0 +1,57 @@
+﻿using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+using BaGet.Decompiler.Nupkg.Objects;
+using BaGet.Decompiler.Objects;
+using NuGet.Packaging;
+
+namespace BaGet.Decompiler.Nupkg
+{
+    public class NugetDecompilerService
+    {
+        private readonly AssemblyDecompilerService _assemblyDecompilerService;
+
+        public NugetDecompilerService(AssemblyDecompilerService assemblyDecompilerService)
+        {
+            _assemblyDecompilerService = assemblyDecompilerService;
+        }
+
+        public AnalysisNugetPackage AnalyzePackage(PackageArchiveReader package)
+        {
+            // TODO: Read pdb's
+            // TODO: Read xml docs
+
+            var res = new AnalysisNugetPackage();
+
+            foreach (var item in package.GetReferenceItems())
+            {
+                // TODO: Can a reference be multiple files?
+                var file = item.Items.Single();
+                var folder = file.Substring(0, file.Length - Path.GetFileName(file).Length - 1);
```

Comment:
I think you can use `Path.GetDirectoryName` here

Reply:
Nope. It turns the slashes.